compiler/rustc_builtin_macros/src/util.rs

```diff
@@ -10,3 +12,28 @@ pub fn check_builtin_macro_attribute(ecx: &ExtCtxt<'_>, meta_item: &MetaItem, na
     let attr = ecx.attribute(meta_item.clone());
     validate_attr::check_builtin_attribute(&ecx.sess.parse_sess, &attr, name, template);
 }
+
+/// Emit a warning if the item is annotated with the given attribute. This is used to diagnose when
+/// an attribute may have been mistakenly duplicated.
+pub fn warn_on_duplicated_attribute(session: &Session, item: &Annotatable, name: Symbol) {
```

Comment:
```suggestion
pub fn warn_on_duplicate_attribute(session: &Session, item: &Annotatable, name: Symbol) {
```
https://english.stackexchange.com/questions/67463/duplicate-data-or-duplicated-data